src/analysis.py

```diff
@@ -0,0 +1,219 @@
+"""
+This script trains models and output results in the form of a figure to be used for further analysis.
+
+Usage: analysis.py -i=<input> -o=<output> [-v]
+
+Options:
+-i <input>, --input <input>     Local raw data csv file directory
+-o <output>, --output <output>  Local output filename and path for preprocessed csv
+[-v]                            Report verbose output of dataset retrieval process
+"""
+
+import os
+import re
+import sys
+from hashlib import sha1
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from sklearn import datasets
+from sklearn.compose import (
+    ColumnTransformer,
+    TransformedTargetRegressor,
+    make_column_transformer,
+)
+from sklearn.dummy import DummyClassifier, DummyRegressor
+from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
+from sklearn.impute import SimpleImputer
+from sklearn.linear_model import LogisticRegression, Ridge, LinearRegression
+from sklearn.multiclass import OneVsRestClassifier
+from sklearn.tree import DecisionTreeClassifier
+from sklearn.svm import SVC
+from sklearn.metrics import (
+    accuracy_score,
+    classification_report,
+    confusion_matrix,
+    f1_score,
+    make_scorer,
+    precision_score,
+    recall_score,
+)
+from sklearn.model_selection import (
+    GridSearchCV,
+    RandomizedSearchCV,
+    cross_validate,
+    train_test_split,
+)
+from sklearn.pipeline import Pipeline, make_pipeline
+from sklearn.preprocessing import OneHotEncoder, OrdinalEncoder, StandardScaler
+from sklearn.svm import SVC, SVR
+
+import altair as alt
+from vega_datasets import data
+
+from docopt import docopt
+from pylab import savefig
+from render_table import render_table
+
+print("start")
+args = docopt(__doc__)
+
+def read_input_file(input_file_path):
```

Comment:
I can modify later to update, but we may eventually want to split this into a tests validation function.

Reply:
Yes, think there is room for improvement WRT to modularization.

Reply:
Haha, low priority at this point for sure.